specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/subAssessments.json

```diff
@@ -236,87 +236,19 @@
           "description": "The date and time the sub-assessment was generated"
         },
         "resourceDetails": {
-          "$ref": "#/definitions/ResourceDetails"
+          "$ref": "../../../common/v1/types.json#/definitions/ResourceDetails"
         },
         "additionalData": {
           "$ref": "#/definitions/AdditionalData"
         }
       }
     },
-    "ResourceDetails": {
-      "type": "object",
-      "description": "Details of the resource that was assessed",
-      "discriminator": "source",
-      "properties": {
-        "source": {
-          "readOnly": true,
-          "type": "string",
-          "description": "The platform where the assessed resource resides",
-          "enum": [
-            "Azure",
-            "Aws"
-          ],
-          "x-ms-enum": {
-            "name": "Source",
-            "modelAsString": true,
-            "values": [
-              {
-                "value": "Azure"
-              },
-              {
-                "value": "Aws"
-              }
-            ]
-          }
-        }
-      }
-    },
-    "AzureResourceDetails": {
-      "type": "object",
-      "description": "Details of the resource that was assessed",
-      "x-ms-discriminator-value": "Azure",
-      "allOf": [
-        {
-          "$ref": "#/definitions/ResourceDetails"
-        }
-      ],
-      "properties": {
-        "id": {
-          "readOnly": true,
-          "type": "string",
-          "description": "Azure resource ID of the assessed resource"
-        }
-      }
-    },
-    "AwsResourceDetails": {
```

Comment:
what happened to this type?

Reply:
the definition moved from here to a common file so it can be used by the "assessments" and "subAssessments" types without duplication
they have to be the same object always by definition

Reply:
AwsResourceDetails was removed completely and not added to another file, was that intended?

Reply:
yes, the AWS resource details is not public yet so it was removed
we will add it once it will be public and the structure will be more stable